R/summary-log.R

```diff
@@ -266,3 +281,120 @@ do_if_bbi_sum <- function(fn, mode) {
     return(res)
   }
 }
+
+#' Calculate Akaike Information Criterion (AIC) and Bayesian Information
+#' Criterion (BIC), and append to summary log
+#' @param .log_df The output of [summary_log()]) or `run_log() %>% add_summary()`
+#' @keywords internal
+add_aic_bic <- function(.log_df){
+  .log_df %>% dplyr::mutate(
+    !!rlang::sym(AIC_COL) := 2*.data[[PARAM_COUNT_COL]] + .data[[OFV_COL]],
+    !!rlang::sym(BIC_COL) := .data[[PARAM_COUNT_COL]]*log(.data$number_of_obs) + .data[[OFV_COL]]
+  ) %>% dplyr::relocate(all_of(c(AIC_COL, BIC_COL)), .after = all_of(OFV_COL))
+}
+
+
+#' Calculate the change objective function value models in a run log
+#'
+#' Extend a run log with a "dofv" column that reports the difference in
+#' objective function value between row's model and a reference model.
+#'
+#' @param .log_df A `bbi_run_log_df` tibble (as output by [run_log()] or
+#'   [summary_log()]).
+#' @param .mod_ref The reference model to use when calculating the change in
+#'   objective function value for each model in the run log. By default, the
+#'   reference model is each model's parent (as identified by the first value
+#'   returned by [get_based_on()]). Pass a `bbi_nonmem_model` object to use its
+#'   objective function value as the reference value for all calculations.
+#' @return The `.log_df` data frame with a new "dofv" column. If `.log_df` is
+#'   not a summary log, a "ofv" column is also added for context.
+#'
+#' @examples
+#' \dontrun{
+#'
+#' # Append to run logs:
+#' run_log(MODEL_DIR) %>% add_summary() %>% add_dofv()
+#' summary_log(MODEL_DIR) %>% add_dofv()
+#'
+#' # Automatically adds `ofv` column in absence of `add_summary()` call:
+#' run_log(MODEL_DIR) %>% add_dofv()
+#' }
+#'
+#'
+#' @seealso [add_summary()]
+#' @export
+add_dofv <- function(.log_df, .mod_ref = NULL) {
```

Comment:
I'll push an update to restore the check that "absolute_model_path" is present as a column.